vault/quotas/quotas.go

```diff
@@ -165,7 +165,15 @@ type Manager struct {
 
 	logger     log.Logger
 	metricSink *metricsutil.ClusterMetricSink
-	lock       *sync.RWMutex
+
+	// quotaLock is a lock for manipulating quotas
+	quotaLock *sync.RWMutex
+
+	// rateLimitExemptPathsLock is a lock specifically for accessing RateLimitExemptPaths
+	rateLimitExemptPathsLock *sync.RWMutex
```

Comment:
Would it make sense to extend this to cover all config-modifying operations, i.e rename this to configLock?  Could we merge it with dbAndCacheLock?

Reply:
We could merge it, but it wouldn't make sense to me as the times these are written to are somewhat unrelated. I like the idea of config lock though, so I've changed the name - thanks!

Reply:
It's mostly about simplifying reasoning about cases where we need to grab multiple locks.